blog/20220404-simplex-chat-instant-notifications.md

```diff
@@ -0,0 +1,124 @@
+
+## Problem - users expect to be instantly notified when messages arrive!
+
+Once our first users started using the apps they realized that what we take for granted in messaging apps - instant message notifications - is missing in our first release of SimpleX Chat apps. Quite a few people thought that it was a bug, rather than a missing feature - sorry to disappoint!
+
+Most messaging apps appear to us very simple, even minimalist, but some features require complex engineering – so we didn't manage to include instant notifications in the first release – and it seems like it made our apps not viable for many users.
+
+## Why it is a bigger problem for SimpleX Chat - and why can't we just do what messenger X does?
+
+SimpleX Chat appears to be the first and the only messenger that operates without user identities of any kind - there are no phone numbers, emails, usernames, public keys, or any other addresses or identifiers to uniquely identify the users to the network or servers - that is why we say it is 100% private by design. Many people believe that to organize communications the servers have to know who the users are, but SimpleX protocol manages message routing without user identities – it authorizes the user actions without authenticating the users.
+
+Instead, SimpleX Chat assigns these identifiers to unidirectional message queues, and what looks to SimpleX Chat users as contacts and groups [1], to SimpleX servers looks like an unorganized and unrelated collection of unidirectional message queues - servers do not know which queues belong to which users, contacts or groups, and even a single conversation can happen via two different servers (one for sent and another for received messages). This makes our personal network of contacts invisible to the servers.
+
+But it also creates a problem for instant notifications - all push notification services require having a unique device token.
+
+So how do we reconcile it, or is it simply impossible both operating without identities and having instant notifications?
+
+[1] yes, we have groups in our terminal app, and the UI to manage them is coming to mobile apps soon - some users have already firgured out how to [create groups via chat console](https://medium.com/@vsevolod.mineev/how-to-collaborate-across-multiple-devices-whilst-protecting-your-metadata-371af87d0ba0).
+
+## What we did for Android
+
+After some research of how push notifications work on Android, and what open-source altetnatives exist to Google push notifications, it turned out to be possible to avoid sharing any device tokens with any servers.
```

Comment:
```suggestion
After some research of how push notifications work on Android, and what open-source alternatives exist to Google push notifications, it turned out to be possible to avoid sharing any device tokens with any servers.
```